OpenRA.Mods.Common/Activities/Attack.cs

```diff
@@ -64,6 +64,14 @@ protected virtual Activity InnerTick(Actor self, AttackBase attack)
 			if (!attack.Info.IgnoresVisibility && type == TargetType.Actor && !Target.Actor.HasTrait<FrozenUnderFog>() && !self.Owner.CanTargetActor(Target.Actor))
 				return NextActivity;
 
+			// Drop the target if it is being captured by our own team.
+			// Cannot check for sabotage-in-progress without reference inside capturable object to the object doing the capturing.
+			var capturable = Target.Actor.TraitOrDefault<Capturable>();
+			var externalcapturable = Target.Actor.TraitOrDefault<ExternalCapturable>();
+			
+			if ((capturable != null || externalcapturable != null) && (capturable.BeingCaptured || externalcapturable.CaptureInProgress))
```

Comment:
This will cause null reference exceptions this way as you don't chain the null check to the corresponding field.

```csharp
if ((capturable != null && capturable.BeingCaptured) || (externalcapturable != null && externalcapturable.CaptureInProgress))
```

is what you want here. Also far more readable.